modules/usb-gadget.nix

```diff
@@ -0,0 +1,9 @@
+{ config, lib, pkgs, ... }:
+
+{
+  # Ensure configfs is mounted if needed.
+  fileSystems."/sys/kernel/config" = lib.mkIf (config.mobile.usb.mode == "gadgetfs") {
```

Comment:
This differs from the original commit in that we're putting the mountpoint location in the attrset name.

This is better since it means that if any other modules defines this mountpoint, it either conflicts or is a no-op.